! # Tutorial Video Walkthrough
! # video_walkthrough_ai_chatgpt_usages.diff

- TITLE
+ Using ChatGPT on a daily work as a P.O, Developer or for Q/A or Support and checking plagiarism if needed with Python
--- See more on the post at https://wp.me/p3Vuhl-3dF

- URLs
+ ARTICLE :: https://wp.me/p3Vuhl-3dF
+ CODE :: https://bit.ly/3YI5Ubl
+ WEBSITE ::  https://flaven.fr/

- QUICK DESCRIPTION

--- OBJECTIVE :: How to use ChatGPT to replace tedious work by AI + checking plagiarism if needed with Python

--- OUPUT :: Numerous example from Jira ticket for P.O, Q/A testing script in JS, WP plugin in PHP, Pyhton stuff, SQL code...

--- AUDIENCE :: Anyone who want to use ChatGPT on a daily basis.


- TEK GUIDELINES
--- ChatGPT :: https://chat.openai.com
--- Streamlit :: https://streamlit.io/
--- scikit-learn :: hhttps://scikit-learn.org/stable/index.html


- PATH CONSOLE
cd /Users/brunoflaven/Documents/03_git/BlogArticlesExamples/ai_chatgpt_usages/


- CONDA ENVIRONMENT
--- requirement using conda to manage development environment
--- create requirements file for the apps
pip freeze > requirements_plagiarism_checker_python.txt
pip install -r requirements_plagiarism_checker_python.txt


! CONTENT

+ 1 + TWO REMARKS ON CHATGPT
First, asking Chat GPT about moral issues is pointless and sometimes ludicrous!

Second, still about the "corpus", as we do not know exactly the corpus of texts on which ChatGPT was trained to perfect its learning, it is certainly obvious that all the documentation relating to the development languages has been included (SQL, PHP, Python, JavaScript and so on... etc.) but that they do not necessarily integrate the latest developments.



+ 2 + HOW TO DETECT PLAGIARISM?
The idea naturally came because of using GPT chat https://chat.openai.com/ is how do we ensure that is has been made by a real person or by an IA?
Consequently, the massive use of GPT Chat or any other form of AI device necessarily brings its corollary of control which comes down to the following question: Is it a human who wrote all or part of the text or code that I have under the eyes?

+ 3 + EXAMPLES GENERATED WITH CHATGPT FOR PRODUCT OWNER, DEVELOPER...

- 1. PO: generate Jira tickets for user stories, tasks, and bugs

! # Write a Jira ticket to create a button that add automatically the author tag to the editing content. The ticket must give a precise DOR (definition of ready) and DOD (definition of done) to clarify the Acceptance Criteria.

- 2. Q/A: Some testing examples

! Write the cypress code to test that the fields: title, body, tags has been properly filled by the user and then save the content created as draft

- 3. DEV: Some Coding examples
--- 1 – Coding: Query example for GPT to generate code for MySQL
! WordPress custom SQL query examples

! # Explain with simple words the following code
global $wpdb;
$wpdb->insert(
    $wpdb->foods,
    array(
        'fruit' => 'apple',
        'year' => 2012
    ),
    array(
        '%s',
        '%d'
    )
);

! # Create table https://github.com/bflaven/BlogArticlesExamples/blob/e0d7c9858285eb547730db268ea0114ef8d5d3a2/extending_streamlit_usage/009_countries_crud_streamlit_app/sqlite_try_3_create_table.py

sql_create_countries_table = """CREATE TABLE IF NOT EXISTS countries (
                                    id INTEGER PRIMARY KEY,
                                    name TEXT NOT NULL,
                                    tld TEXT NOT NULL UNIQUE,
                                    cca2 TEXT NOT NULL UNIQUE,
                                    capital TEXT NOT NULL UNIQUE,
                                    callingCode INTEGER,
                                    FOREIGN KEY (id) REFERENCES countries (id)
                                );"""

--- 2 – Coding: Query example for GPT to generate code in Python
! Write python wit comments using beautyfulsoup, read a file name example_3.html and find out the H5 tag and H6 tag in the HTML? source. The H5 is called a smallBookTitles, it has to be inject in a loop to create a list for all the books showed in the page example_3.html

! # Model: https://github.com/bflaven/BlogArticlesExamples/blob/34fe0f51a53b0ae38af614e66b9eeaf16facf97d/webscraping_with_python/slurpThatSoup_6.py



--- 3 – Coding: Query example for GPT to generate code in PHP for a plugin WordPress named “MySocialBuddy”

! Write a WordPress plugin that enable to manage different languages with flags

! # Write in PHP, a Wordpress plugin with comments, named MyFlagsSiteVersion. This Plugin MyFlagsSiteVersion will add a new entry in the main administration menu in "Settings". The name of the entrance will be MyFlagsSiteVersion. The purpose of this plugin is to manipulate 3 custom fields for different languages with flags accounts: myflagssiteversion_fr, myflagssiteversion_en, myflagssiteversion_sp. You can you write also tips for this field and general explanation at the beginning of the plugin page settings to explain how does the plugin work.

! # Model: https://github.com/bflaven/PluginWordpressForFun/blob/master/johann_flags_header/johann_flags_header.php

- 4 – Coding: Other attempts...


+ 4. DEV: ERRORS EXPLAINED

--- Some errors searchings quicker to look in stackoverflow.com. The error can be explained by ChatGPT, it is complementary source of information with stackoverflow.com for instance

! # sample for errors...
--- Explain the python error "TypeError: Protocols cannot be instantiated"
--- Explain the python error "TypeError: Protocols cannot be instantiated for Streamlit what is the solution?"
--- How to update python in anaconda (conda)?


+ 5. SUPPORT: MAILS FOR SUPPORT
--- 1 - Support: Query example for GPT to generate support mail for 2FA authentication
--- 2 - Support: Query example for GPT to generate support mail for user to explain how-to to login/logout and start private session
--- 3 - Support: Query example for GPT to generate support mail




! EXTRA

- HASHTAGS
An #experimentation and #POC for #plagiarism #checker #ScikitLearn #Python #Streamlit, https://bit.ly/3YI5Ubl

Using #ChatGPT on a #daily #work as a #PO, #Developer, for #QA or #Support and #checking #plagiarism if needed with #Python, https://wp.me/p3Vuhl-3dF

- GIT COMMIT
git status
git branch
git add .
git commit -am "updates files make it work for video"
git commit -am "change on several README files"
git push origin master




